sql/handler.h

```diff
@@ -4840,9 +4843,10 @@ int ha_release_savepoint(THD *thd, SAVEPOINT *sv);
 #ifdef WITH_WSREP
 int ha_abort_transaction(THD *bf_thd, THD *victim_thd, my_bool signal);
 void ha_fake_trx_id(THD *thd);
-#else
+void wsrep_wait_until_innodb_initialized();
+ #else
 inline void ha_fake_trx_id(THD *thd) { }
-#endif
+#endif /* WITH_WSREP */
```

Comment:
Do we really need yet another `handler` member function? Shouldn’t it at least be `static`?*
Code that is specific to a single storage engine should be avoided in `handler`.

Reply:
It is true that handlerton method wait_until_initialized is implemented only for innodb SE, but in principle it should be implemented for all SEs which support wsrep API (and thus can form a cluster), and which may be running conflicting background tasks at startup phase.